### ISSUE TITLE
Quantized model (qnnpack) is slower than floating point counterpart and the output is corrupted

### ISSUE THREAD

**Issue by gp2nmo** [pytorch/pytorch#69671]
## 🐛 Bug

Quantized generator model runs significantly slower than the floating point counterpart. The slowdown is present with both fbgemm and qnnpack. Besides the slowdown, when running the inference with qnnpack (tested on multiple arm android devices), the output is corrupted.

## To Reproduce

1. Clone the repo https://github.com/bryandlee/animegan2-pytorch
2. In `model.py` modify the `Generator`'s forward pass to quantize the input and than dequantize the output
3. In `InvertedResBlock` modify the body of the ` if self.use_res_connect:` statement so that the tensor addition is performed using FloatFunctional.
4. Quantize the model and optimize it for lite inference:

````python
        model_fp32 = QuantizedGenerator()
        model_fp32.load_state_dict(torch.load(args.checkpoint, map_location="cpu"))
        model_fp32.to(device).eval()
        model_fp32.qconfig = torch.quantization.get_default_qconfig('qnnpack')
        model_fp32_prepared = torch.quantization.prepare(model_fp32, inplace=True)
        input_fp32 = torch.randn(1, 3, 512, 512)
        model_fp32_prepared(input_fp32)
        model_int8 = torch.quantization.convert(model_fp32_prepared, inplace=True)
        scripted_int8 = torch.jit.script(model_int8)
        optimized_int8 = mobile_optimizer.optimize_for_mobile(scripted_int8)
        optimized_int8._save_for_lite_interpreter("face_v2_lite_quant.pt")
````

5. Run the model on an ARM Android device.

## Expected behavior

The model should not run slower than the floating point counterpart, and the output should not be corrupted.

[Example input](https://t0.gstatic.com/licensed-image?q=tbn:ANd9GcT27KhuuSKYvAbCNywbEL3W3USsYIHDM-REVYfjY9pmIIwGg3v0LeV0krgFQ5VK)

[Floating point model output](https://pbs.twimg.com/media/FFleQTMX0AM9dbw?format=jpg&name=large)

[qnnpack quantized output](https://ibb.co/hKBtb7n)

(Ignore the cropping)

## Environment

 - PyTorch Version (e.g., 1.0): Model conversion: 1.10.0, Inference: pytorch_android_lite_1.10.0
 - OS (e.g., Linux): Model conversion: Windows 10, Inference: Android
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.9.1
 - CUDA/cuDNN version:  N/A
 - GPU models and configuration:  N/A
 - Any other relevant information:  N/A

## Additional context

 N/A

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo

**Comment by vkuzo** [comment]
Hi @gp2nmo , 

For the slowdown, do you have any on-device profiling results which give a per-op breakdown?  That would help narrow down the issue.

For the accuracy issues, are you calibrating your model with a representative dataset?  In the example shared in the issue, the only calibration I see is with one tensor filled with random noise, this is expected to give incorrect results.  Have you tried calibrating your model with ~100-500 images which are representative of what you plan to see at inference?

Also, we are using the issues for bug reports, questions like this one we are funneling to our discuss forum (https://discuss.pytorch.org/c/quantization/17).  Please feel free to post any answers to these questions or any follow-ups there and we will be glad to help.  Closing this issue since this is not a bug report.
